src/redux/books/booksSlice.js

```diff
@@ -1,50 +1,90 @@
-import { createSlice } from '@reduxjs/toolkit';
-import produce from 'immer';
-
-const bookItems = [
-  {
-    id: 'item1',
-    title: 'The Great Gatsby',
-    author: 'John Smith',
-    category: 'Fiction',
-  },
-  {
-    id: 'item2',
-    title: 'Anna Karenina',
-    author: 'Leo Tolstoy',
-    category: 'Fiction',
-  },
-  {
-    id: 'item3',
-    title: 'The Selfish Gene',
-    author: 'Richard Dawkins',
-    category: 'Nonfiction',
-  },
-];
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import axios from 'axios';
+
+const baseUrl = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/';
+const urlEndPoint = 'z2NP5NXo87KPEKUYTK76/books/';
+const apiUrl = baseUrl + urlEndPoint;
+
+function transformData(data) {
+  return data.map(([id, [book]]) => {
+    const { title, author, category } = book;
+    return {
+      id, title, author, category,
+    };
+  });
+}
+
+export const fetchBooks = createAsyncThunk('books/fetchBooks', async () => {
+  const response = await axios.get(apiUrl);
+  const { data } = response;
+  return transformData(Object.entries(data));
+});
+
+export const addBook = createAsyncThunk('books/addBook', async ({ id, title, author }) => {
+  const response = await axios.post(apiUrl, {
+    item_id: id,
+    title,
+    author,
+    category: 'religion',
+  }, {
+    headers: {
+      'Content-Type': 'application/json',
+    },
+  });
+  return response.data;
+});
+
+export const removeBook = createAsyncThunk('books/removeBook', async (bookId) => {
+  await fetch(`${apiUrl}/${bookId}`, {
+    method: 'DELETE',
+  });
```

Comment:
- [ ] You have done a great job @Leeoasis, But at this point, your `removeBook` functionality is not working. It removes the book from the browser successfully but fails to delete it from the API endpoint. Please address this to make your app functional as expected.
> Take a look at this, please.
[Screencast from 06-04-2023  3:28:54 ከሰዓት.webm](https://user-images.githubusercontent.com/69077061/230380072-18b195d0-25f8-4acd-abb9-d7a706233842.webm)